src/apis/impl/AuthenticationApiImpl.ts

```diff
@@ -135,19 +152,19 @@ export class AuthenticationApiImpl implements AuthenticationApi {
           delegate
         );
         //TODO Return which delegates were warned to share back info & add retry mechanism
-        if (!accessNotification) throw new Error(`Cannot create a notification to Healthcare Professional with id ${delegate}`)
+        if (!accessNotification) throw this.errorHandler.createErrorWithMessage(`Cannot create a notification to Healthcare Professional with id ${delegate}`)
       }
 
     } else if (!!loggedUser.healthcarePartyId) {
       const hcpDataOwner = await authenticationResult.medTechApi.healthcareProfessionalApi.getHealthcareProfessional(loggedUser.healthcarePartyId);
-      if (!hcpDataOwner) throw new Error(`Healthcare Professional with id ${loggedUser.healthcarePartyId} does not exist`);
+      if (!hcpDataOwner) throw this.errorHandler.createErrorWithMessage(`Healthcare Professional with id ${loggedUser.healthcarePartyId} does not exist`);
 
     } else if (!!loggedUser.deviceId) {
       const hcpDataOwner = await authenticationResult.medTechApi.medicalDeviceApi.getMedicalDevice(loggedUser.deviceId);
-      if (!hcpDataOwner) throw new Error(`Medical Device with id ${loggedUser.deviceId} does not exist`);
+      if (!hcpDataOwner) throw this.errorHandler.createErrorWithMessage(`Medical Device with id ${loggedUser.deviceId} does not exist`);
 
     } else {
-      throw new Error("User is not a Data Owner")
+      throw this.errorHandler.createErrorWithMessage(`User with id ${loggedUser.id} is not Data Owner (either a patient, a healthcare professional or a medical device)`)
```

Comment:
`User with id ${loggedUser.id} is not a Data Owner. To be a Data Owner, your user needs to have either patientId, healthcarePartyId or deviceId filled`